source/grid/grid_generator.cc

```diff
@@ -7250,29 +7363,101 @@ namespace GridGenerator
         // create cells one by one
 
         if (dim == 2)
-          for (const auto &cell : table_2D_cell)
-            add_cell(dim, cell, current_material_id);
+          {
+            const auto current_manifold_id_of_cell =
+              cell.manifold_id(); // get cell-manifold id from current quad cell
+            for (const auto &cell : table_2D_cell)
```

Comment:
This is pre-existing, but it's confusing nonetheless. In line 7150 (old line number) above, you already have
```
 for (const auto &cell : in_tria)
```
so you have a local variable `cell` that is a `cell_iterator`. Here, you introduce another local variable `cell` of a different data type that is not actually a cell but (I believe) a triplet of numbers. Choose a different variable name to avoid this kind of confusion. Try to reflect in the variable name what the triplet actually represents.

Reply:
I totally agree to change those variable names!